### PR TITLE
Add Alicia Cook keyboard

### PR DESCRIPTION
**Pull request by ibnuda**
add support for a split alice that will be given away soon in r/emk.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/alicia_cook/readme.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Alicia Cook
+
+![Alicia Cook](https://raw.githubusercontent.com/ibnuda/alicia-cook/master-of-puppets/alicia-cook.jpg)
+
+*A short description of the keyboard/project*
```
Comment (zvecr) [inline keyboards/alicia_cook/readme.md:5]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/alicia_cook/info.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
+{
+  "keyboard_name": "Alicia Cook",
+  "url": "https://github.com/ibnuda/alicia-cook",
+  "maintainer": "ibnuda",
+  "width": 13,
+  "height": 4,
+  "layouts": {
+      "LAYOUT": {
+          "layout": [
+            {"x":0,"y":0,"w":1,"h":1},
```
Comment (fauxpark) [inline keyboards/alicia_cook/info.json:10]: Width and height for 1u keys can be omitted.

File: keyboards/alicia_cook/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        KC_ESC, KC_1,   KC_2,   KC_3,   KC_4,   KC_5,            KC_6,   KC_7,   KC_8,   KC_9,   KC_0,   KC_DEL, KC_BSPC, \
+        KC_LCTL,KC_EXLM,KC_AT,  KC_HASH,KC_DLR, KC_PERC,         KC_CIRC,KC_AMPR,KC_ASTR,KC_LPRN,KC_RPRN,   KC_SCLN,      \
+        KC_LSFT,KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,            KC_N,   KC_M,   KC_COMM,KC_DOT, KC_SLSH,KC_RSFT,MO(2),   \
+        KC_LGUI,KC_LALT,        KC_LGUI,LW_BSPC,SFT_ESC,         ALT_ENT,RS_SPC, KC_LALT,                KC_RGUI,KC_RCTL  \
+    ),
+    [2] = LAYOUT(
+        KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,           KC_F7,  KC_F8,  KC_F9,  KC_F10, KC_F11, KC_F12, KC_F13, \
+        KC_LCTL,KC_EXLM,KC_AT,  KC_HASH,KC_DLR, KC_PERC,         KC_CIRC,KC_AMPR,KC_ASTR,KC_LPRN,KC_RPRN,   KC_SCLN,     \
+        KC_LSFT,KC_Z,   KC_X,   KC_C,   KC_V,   RESET,           KC_N,   KC_M,   KC_COMM,KC_DOT, KC_SLSH,KC_RSFT,MO(1),  \
+        KC_LGUI,KC_LALT,        KC_LGUI,LW_BSPC,SFT_ESC,         ALT_ENT,RS_SPC, KC_LALT,                KC_RGUI,KC_RCTL \
```
Comment (fauxpark) [inline keyboards/alicia_cook/keymaps/default/keymap.c:53]:
```suggestion
        KC_ESC, KC_Q,   KC_W,   KC_E,   KC_R,   KC_T,            KC_Y,   KC_U,   KC_I,   KC_O,   KC_P,   KC_DEL, KC_BSPC,
        KC_LCTL,SH_A,   CT_S,   AL_D,   KC_F,   KC_G,            KC_H,   KC_J,   KC_K,   KC_L,   KC_QUOT,     KC_SCLN,
        KC_LSFT,KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,            KC_N,   KC_M,   KC_COMM,KC_DOT, KC_SLSH,KC_RSFT,MO(1),
        KC_LGUI,KC_LALT,        KC_LGUI,LW_BSPC,SFT_ESC,         ALT_ENT,RS_SPC, KC_LALT,                KC_RGUI,KC_RCTL
    ),
    [1] = LAYOUT(
        KC_ESC, KC_1,   KC_2,   KC_3,   KC_4,   KC_5,            KC_6,   KC_7,   KC_8,   KC_9,   KC_0,   KC_DEL, KC_BSPC,
        KC_LCTL,KC_EXLM,KC_AT,  KC_HASH,KC_DLR, KC_PERC,         KC_CIRC,KC_AMPR,KC_ASTR,KC_LPRN,KC_RPRN,   KC_SCLN,
        KC_LSFT,KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,            KC_N,   KC_M,   KC_COMM,KC_DOT, KC_SLSH,KC_RSFT,MO(2),
        KC_LGUI,KC_LALT,        KC_LGUI,LW_BSPC,SFT_ESC,         ALT_ENT,RS_SPC, KC_LALT,                KC_RGUI,KC_RCTL
    ),
    [2] = LAYOUT(
        KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,           KC_F7,  KC_F8,  KC_F9,  KC_F10, KC_F11, KC_F12, KC_F13,
        KC_LCTL,KC_EXLM,KC_AT,  KC_HASH,KC_DLR, KC_PERC,         KC_CIRC,KC_AMPR,KC_ASTR,KC_LPRN,KC_RPRN,   KC_SCLN,
        KC_LSFT,KC_Z,   KC_X,   KC_C,   KC_V,   RESET,           KC_N,   KC_M,   KC_COMM,KC_DOT, KC_SLSH,KC_RSFT,MO(1),
        KC_LGUI,KC_LALT,        KC_LGUI,LW_BSPC,SFT_ESC,         ALT_ENT,RS_SPC, KC_LALT,                KC_RGUI,KC_RCTL
```

File: keyboards/alicia_cook/keymaps/rick/keymap.c [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        KC_ESC, KC_1,   KC_2,   KC_3,   KC_4,   KC_5,            KC_6,   KC_7,   KC_8,   KC_9,   KC_0,   KC_DEL, KC_BSPC, \
+        KC_LCTL,KC_EXLM,KC_AT,  KC_HASH,KC_DLR, KC_PERC,         KC_CIRC,KC_AMPR,KC_ASTR,KC_LPRN,KC_RPRN,   KC_SCLN,      \
+        KC_LSFT,KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,            KC_N,   KC_M,   KC_COMM,KC_DOT, KC_SLSH,KC_RSFT,MO(2),   \
+        KC_LGUI,KC_LALT,        KC_LGUI,LW_BSPC,SFT_ESC,         ALT_ENT,RS_SPC, KC_LALT,                KC_RGUI,KC_RCTL  \
+    ),
+    [2] = LAYOUT(
+        KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,           KC_F7,  KC_F8,  KC_F9,  KC_F10, KC_F11, KC_F12, KC_F13, \
+        KC_LCTL,KC_EXLM,KC_AT,  KC_HASH,KC_DLR, KC_PERC,         KC_CIRC,KC_AMPR,KC_ASTR,KC_LPRN,KC_RPRN,   KC_SCLN,     \
+        KC_LSFT,KC_Z,   KC_X,   KC_C,   KC_V,   RESET,           KC_N,   KC_M,   KC_COMM,KC_DOT, KC_SLSH,KC_RSFT,MO(1),  \
+        KC_LGUI,KC_LALT,        KC_LGUI,LW_BSPC,SFT_ESC,         ALT_ENT,RS_SPC, KC_LALT,                KC_RGUI,KC_RCTL \
```
Comment (fauxpark) [inline keyboards/alicia_cook/keymaps/rick/keymap.c:53]:
```suggestion
        KC_ESC, KC_SCLN,KC_COMM,KC_DOT, KC_P,   KC_Y,            KC_F,   KC_G,   KC_C,   KC_R,   KC_L,   KC_BSPC, KC_BSPC,
        KC_LCTL,SH_A,   CT_O,   AL_T,   KC_U,   GU_I,            GU_D,   KC_H,   AL_T,   CT_N,   SH_S,   KC_ENT,
        KC_LSFT,KC_QUOT,KC_Q,   KC_J,   KC_K,   KC_X,            KC_B,   KC_M,   KC_W,   KC_V,   KC_Z,   KC_RSFT,MO(1),
        KC_LGUI,KC_LALT,        KC_LGUI,LW_BSPC,SFT_ESC,         ALT_ENT,RS_SPC, KC_LALT,                KC_RGUI,KC_RCTL
    ),
    [1] = LAYOUT(
        KC_ESC, KC_1,   KC_2,   KC_3,   KC_4,   KC_5,            KC_6,   KC_7,   KC_8,   KC_9,   KC_0,   KC_DEL, KC_BSPC,
        KC_LCTL,KC_EXLM,KC_AT,  KC_HASH,KC_DLR, KC_PERC,         KC_CIRC,KC_AMPR,KC_ASTR,KC_LPRN,KC_RPRN,   KC_SCLN,
        KC_LSFT,KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,            KC_N,   KC_M,   KC_COMM,KC_DOT, KC_SLSH,KC_RSFT,MO(2),
        KC_LGUI,KC_LALT,        KC_LGUI,LW_BSPC,SFT_ESC,         ALT_ENT,RS_SPC, KC_LALT,                KC_RGUI,KC_RCTL
    ),
    [2] = LAYOUT(
        KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,           KC_F7,  KC_F8,  KC_F9,  KC_F10, KC_F11, KC_F12, KC_F13,
        KC_LCTL,KC_EXLM,KC_AT,  KC_HASH,KC_DLR, KC_PERC,         KC_CIRC,KC_AMPR,KC_ASTR,KC_LPRN,KC_RPRN,   KC_SCLN,
        KC_LSFT,KC_Z,   KC_X,   KC_C,   KC_V,   RESET,           KC_N,   KC_M,   KC_COMM,KC_DOT, KC_SLSH,KC_RSFT,MO(1),
        KC_LGUI,KC_LALT,        KC_LGUI,LW_BSPC,SFT_ESC,         ALT_ENT,RS_SPC, KC_LALT,                KC_RGUI,KC_RCTL
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/alicia_cook/config.h [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+//#    define RGBLIGHT_EFFECT_RGB_TEST
+//#    define RGBLIGHT_EFFECT_ALTERNATING
+/*== customize breathing effect ==*/
+/*==== (DEFAULT) use fixed table instead of exp() and sin() ====*/
+//#    define RGBLIGHT_BREATHE_TABLE_SIZE 256      // 256(default) or 128 or 64
+/*==== use exp() and sin() ====*/
+//#    define RGBLIGHT_EFFECT_BREATHE_CENTER 1.85  // 1 to 2.7
+//#    define RGBLIGHT_EFFECT_BREATHE_MAX    255   // 0 to 255
+//#endif
+
```
Comment (noroadsleft) [inline keyboards/alicia_cook/config.h:92]: All of this can go.

```suggestion
```

File: keyboards/alicia_cook/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+#define CT_L    RCTL_T(KC_L)
+#define SH_A    RSFT_T(KC_A)
+#define SH_SCLN RSFT_T(KC_SCLN)
+#define AL_D    RALT_T(KC_D)
+#define AL_K    RALT_T(KC_K)
+#define GU_G    RGUI_T(KC_G)
+#define GU_H    RGUI_T(KC_H)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/alicia_cook/keymaps/default/keymap.c:37]:
```suggestion
    [0] = LAYOUT_all(
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/11722/files).

**Comment by tzarc** [comment]
Removed tags preventing stalebot from auto-closing this PR.
I'd recommend working through the suggestions if you are keen on getting this PR merged.

**Comment by ibnuda** [comment]
chief. any update?

File: keyboards/alicia_cook/rules.mk [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = no       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/alicia_cook/rules.mk:18]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/alicia_cook/rules.mk [hunk 8]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/alicia_cook/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/alicia_cook/rules.mk [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+EXTRAKEY_ENABLE = no       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/alicia_cook/rules.mk:21]:
```suggestion
```

File: keyboards/alicia_cook/info.json [hunk 10]
```diff
@@ -0,0 +1,64 @@
+{
+  "keyboard_name": "Alicia Cook",
+  "url": "https://github.com/ibnuda/alicia-cook",
+  "maintainer": "ibnuda",
+  "width": 13,
+  "height": 4,
```
Comment (fauxpark) [inline keyboards/alicia_cook/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/alicia_cook/rules.mk [hunk 11]
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/alicia_cook/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
```

File: keyboards/alicia_cook/rules.mk [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = no       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # Enable N-Key Rollover
```
Comment (fauxpark) [inline keyboards/alicia_cook/rules.mk:18]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/alicia_cook/keymaps/rick/config.h [hunk 13]
```diff
@@ -0,0 +1,8 @@
+#pragma once
```
Comment (drashna) [inline keyboards/alicia_cook/keymaps/rick/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/alicia_cook/readme.md [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+
+![Alicia Cook](https://raw.githubusercontent.com/ibnuda/alicia-cook/master-of-puppets/alicia-cook.jpg)
+
+40% Alice-style keyboard that can be split and uses IDC connector.
+
+* Keyboard Maintainer: [Ibnu D. Aji](https://github.com/ibnuda)
+* Hardware Supported: [Alicia Cook](https://github.com/ibnuda/alicia-cook)
+* Hardware Availability: [Alicia Cook](https://github.com/ibnuda/alicia-cook)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/alicia_cook/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/alicia_cook/rules.mk [hunk 15]
```diff
@@ -7,16 +7,15 @@ BOOTLOADER = caterina
 # Build Options
 #   change yes to no to disable
 #
-BOOTMAGIC_ENABLE = no     # Virtual DIP switch configuration
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
```
Comment (noroadsleft) [inline keyboards/alicia_cook/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
```

File: keyboards/alicia_cook/rules.mk [hunk 16]
```diff
@@ -7,16 +7,15 @@ BOOTLOADER = caterina
...
 MOUSEKEY_ENABLE = no       # Mouse keys
 EXTRAKEY_ENABLE = no       # Audio control and System control
 CONSOLE_ENABLE = no         # Console for debug
 COMMAND_ENABLE = no         # Commands for debug and configuration
 # Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
 SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
 # if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (noroadsleft) [inline keyboards/alicia_cook/rules.mk:17]:
```suggestion
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**